sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuAvroScan.scala

```diff
@@ -151,92 +174,322 @@ case class GpuAvroPartitionReaderFactory(
 
   override def buildColumnarReader(partFile: PartitionedFile): PartitionReader[ColumnarBatch] = {
     val conf = broadcastedConf.value.value
-    val blockMeta = GpuAvroFileFilterHandler(sqlConf, broadcastedConf,
-      ignoreExtension, broadcastedConf.value.value).filterBlocks(partFile)
-    val reader = new PartitionReaderWithBytesRead(new AvroPartitionReader(conf, partFile, blockMeta,
-      readDataSchema, debugDumpPrefix, maxReadBatchSizeRows,
+    val blockMeta = AvroFileFilterHandler(conf, options).filterBlocks(partFile)
+    val reader = new PartitionReaderWithBytesRead(new GpuAvroPartitionReader(conf, partFile,
+      blockMeta, readDataSchema, debugDumpPrefix, maxReadBatchSizeRows,
       maxReadBatchSizeBytes, metrics))
     ColumnarPartitionReaderWithPartitionValues.newReader(partFile, reader, partitionSchema)
   }
 }
 
 /**
- * A tool to filter Avro blocks
- *
- * @param sqlConf         SQLConf
- * @param broadcastedConf the Hadoop configuration
+ * The multi-file partition reader factory for cloud or coalescing reading of avro file format.
  */
-private case class GpuAvroFileFilterHandler(
+case class GpuAvroMultiFilePartitionReaderFactory(
     @transient sqlConf: SQLConf,
+    @transient rapidsConf: RapidsConf,
     broadcastedConf: Broadcast[SerializableConfiguration],
-    ignoreExtension: Boolean,
-    hadoopConf: Configuration) extends Arm with Logging {
+    dataSchema: StructType,
+    readDataSchema: StructType,
+    partitionSchema: StructType,
+    options: AvroOptions,
+    metrics: Map[String, GpuMetric],
+    filters: Array[Filter],
+    queryUsesInputFile: Boolean)
+  extends MultiFilePartitionReaderFactoryBase(sqlConf, broadcastedConf, rapidsConf) {
 
-  def filterBlocks(partFile: PartitionedFile): AvroBlockMeta = {
+  private val debugDumpPrefix = Option(rapidsConf.avroDebugDumpPrefix)
+  private val ignoreMissingFiles = sqlConf.ignoreMissingFiles
+  private val ignoreCorruptFiles = sqlConf.ignoreCorruptFiles
+
+  private val numThreads = rapidsConf.avroMultiThreadReadNumThreads
+  private val maxNumFileProcessed = rapidsConf.maxNumAvroFilesParallel
```

Comment:
This is unreferenced

Reply:
Removed

Reply:
It is OK since it is used only by the multi-threaded reader, which will come later, along with this config.

Reply:
Then why does this PR add a config that is unused?  If something goes wrong and we don't ship the multithreaded reader PR before we ship 22.06, then we're advertising a config that doesn't do anything despite its name.  This PR should not be adding code that isn't used.

Reply:
Yes, you are right. It is my mistake missing to remove it after copy some code from the `GpuOrcScan`.